pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap.go

```diff
@@ -319,3 +325,97 @@ func isLinuxGuest(guestID string) bool {
 
 	return strings.Contains(guestID, "Linux") || strings.Contains(guestID, "linux")
 }
+
+func logConfigSpec(
+	vmCtx context.VirtualMachineContextA2,
+	configSpec *vimTypes.VirtualMachineConfigSpec) {
+
+	if !pkgconfig.FromContext(vmCtx).LogFullBootstrapConfigAndCustomizationSpecs {
+		configSpec = SanitizeConfigSpec(configSpec)
+	}
+
+	vmCtx.Logger.Info("Customization Reconfigure", "configSpec", configSpec)
+}
+
+func SanitizeConfigSpec(configSpec *vimTypes.VirtualMachineConfigSpec) *vimTypes.VirtualMachineConfigSpec {
+	cs := *configSpec
+
+	cs.ExtraConfig = slices.Clone(cs.ExtraConfig)
+	for i, ec := range cs.ExtraConfig {
+		optVal := ec.GetOptionValue()
+		if optVal == nil {
+			continue
+		}
+
+		if optVal.Key == constants.CloudInitGuestInfoUserdata {
```

Comment:
It is possible that future metadata could also include sensitive data. Same goes for vendordata (which is 1-to-1 with the userdata format). Please leave a TODO that indicates both.